worker/worker.ts

```diff
@@ -0,0 +1,53 @@
+import { Worker, Job } from 'bullmq';
+import { redisConnection } from '../shared/config';
+
+export interface JobWorker {
+  canProcess(job: any): Promise<boolean>;
+  postpone(job: any): Promise<boolean>;
+  stop(): Promise<void>;
+}
+
+export interface JobProcessor<T> {
```

Comment:
could you just use a function type here ... it's a bit ligheter than creating a class which implements a interface which just has one function?

Reply:
Sure